File: data/src/main/java/hera/database/types/BindingType.java
package hera.database.types;

public enum BindingType {
	MUSIC(0),
	ANNOUNCEMENT(1),
	REPORT(2);

	private int value;

	BindingType(int value) {
		this.value = value;
	}

	public int getValue() {
		return value;
	}
}

File: data/src/main/java/hera/database/entities/BindingType.kt
package hera.database.entities

import hera.database.types.BindingName
import javax.persistence.*

@Entity
@Table(name = "binding_type")
data class BindingType(
		@Id
		@GeneratedValue(strategy = GenerationType.IDENTITY)
		override var id: Long? = null,

		@Enumerated(EnumType.STRING)
		@Column(name = "type")
		var type: BindingName? = null,

		@ManyToOne
		@JoinColumn(name = "localisationFK")
		var message: Localisation? = null,

		var isGlobal: Boolean? = null
) : IPersistenceEntity {
	constructor(type: BindingName, message: Localisation, isOwner: Boolean) : this(null, type, message, isOwner)
}
